tests/Integration/Behaviour/Features/Scenario/Order/order_from_bo.feature

```diff
@@ -333,3 +333,21 @@ Feature: Order from Back Office (BO)
       | Postal code      | 12345                              |
     When I change order "bo_order1" shipping address to "test-address"
     Then order "bo_order1" shipping address should be "test-address"
+
+  Scenario: Generate order then modify product price then add same product on another invoice and check the price
+      Given order "bo_order1" does not have any invoices
+      When I generate invoice for "bo_order1" order
+    When I edit product "Mug The best is yet to come" to order "bo_order1" with following products details:
+      | amount         | 1                       |
+      | price          | 94.33                   |
+      | price_tax_incl | 100                     |
+      | free_shipping  | false                   |
+    When I add products to order "bo_order1" with new invoice and the following products details:
+      | name           | Mug The best is yet to come |
+      | amount         | 1                           |
+      | price          | 94.33                       |
+      | price_tax_incl | 100                         |
+      | free_shipping  | false                       |
+    And product "Mug The best is yet to come" in order "bo_order1" has following details:
+      | unit_price_tax_incl         | 100   |
```

Comment:
You should add tests on other values (like in edition), since it's the only test that checks OrderDetail values on addition (`product_quantity`,  `product_price`,  `total_price_tax_incl`, `total_price_tax_excl`